src/value/index.rs

```diff
@@ -80,12 +80,7 @@ impl Index for str {
         }
         match *v {
             Value::Object(ref mut map) => {
-                // TODO: use entry() once LinkedHashMap supports entry()
-                // https://github.com/contain-rs/linked-hash-map/issues/5
-                if !map.contains_key(self) {
-                    map.insert(self.to_owned(), Value::Null);
-                }
-                map.get_mut(self).unwrap()
+                map.entry(self.to_owned()).or_insert(Value::Null)
```

Comment:
Yes. And previously it did two (sometimes three) traversals to find the key. We are working on a better approach in rust-lang/rfcs#1769 but for now this is the most readable and correct-looking.